javascript/src/seedsource/containers/Map.jsx

```diff
@@ -238,40 +236,38 @@ class Map extends React.Component {
         }
     }
 
-    updateVariableLayer(variable, objective, climate, region) {
-        if (variable !== null) {
-            let layerUrl = '/tiles/' + getServiceName(variable, objective, climate, region) + '/{z}/{x}/{y}.png'
+    updateRasterLayers(layers) {
+        let numLayersToAdd = layers.length - this.displayedRasterLayers.length
+        if (numLayersToAdd === 0) {
+            return
+        }
 
-            if (this.variableLayer === null) {
-                this.variableLayer = L.tileLayer(layerUrl, {zIndex: 1, opacity: 1}).addTo(this.map)
-            }
-            else if(layerUrl !== this.variableLayer._url) {
-                this.variableLayer.setUrl(layerUrl)
+        let rewriteLeafletRasters = () => {
+            if (layers.length) {
+                let { objective, climate, region } = this.props
+                let { serviceId } = this.props.job
+                let url
+                layers.forEach((layer, index) => {
+                    url = getLayerUrl(layer, serviceId, objective, climate, region)
+                    this.displayedRasterLayers[index].setUrl(`/tiles/${url}/{z}/{x}/{y}.png`)
+                        .setZIndex(layer.zIndex)
+                })
             }
         }
-        else if (this.variableLayer !== null) {
-            this.map.removeLayer(this.variableLayer)
-            this.variableLayer = null
-        }
-    }
-
-    updateResultsLayer(serviceId, showResults) {
-        if (serviceId !== null && showResults) {
-            let layerUrl = '/tiles/' + serviceId + '/{z}/{x}/{y}.png'
 
-            if (this.resultsLayer === null) {
-                this.resultsLayer = L.tileLayer(layerUrl, {zIndex: 2, opacity: 1}).addTo(this.map)
+        if (numLayersToAdd > 0) {
+            while (numLayersToAdd > 0) {
+                let newRasterLayer = L.tileLayer("placeHolder", {zIndex: 1, opacity: 1}).addTo(this.map)
+                this.displayedRasterLayers.push(newRasterLayer)
+                numLayersToAdd --
             }
-            else if (layerUrl !== this.resultsLayer._url) {
-                this.resultsLayer.setUrl(layerUrl)
+        } else {
+            while (numLayersToAdd < 0) {
```

Comment:
Similarly, this can be done with `.splice` and `.forEach`:

```javascript
this.displayedRasterLayers
    .splice(numLayersToAdd, Math.abs(numLayersToAdd))
    .forEach(layer => this.map.removeLayer(layer))
```